code/game/objects/items/robot/items/inflatable_tool.dm

```diff
@@ -0,0 +1,41 @@
+/obj/item/inflatable/cyborg
+	name = "inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	var/power_use = 400
+	var/structure_type = /obj/structure/inflatable
+	var/delay = 1 SECONDS
+
+/obj/item/inflatable/cyborg/door
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon_state = "folded_door"
+	power_use = 600
+	structure_type = /obj/structure/inflatable/door
+
+/obj/item/inflatable/cyborg/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>As a synthetic, you can restore them to <b>cyborg recharger</b></span>"
+
+/obj/item/inflatable/cyborg/attack_self(mob/user)
+	if(locate(/obj/structure/inflatable) in get_turf(user))
+		to_chat(user, "<span class='warning'>There's already an inflatable wall!</span>")
+		return FALSE
+
+	if(!do_after(user, delay, FALSE, user))
+		return FALSE
+
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	to_chat(user, "<span class='notice'>You inflate [name]</span>")
+	var/obj/structure/inflatable/R = new structure_type(user.loc)
+	transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	useResource(user)
```

Comment:
This should go above and it should be a check if there is actually enough power to inflate the walls, you can now just make infinite walls if you are almost out of power